assembl/views/api2/auth.py

```diff
@@ -452,21 +475,36 @@ def assembl_register_user(request):
         # Find agent account to avoid duplicates!
         if session.query(AbstractAgentAccount).filter_by(
                 email_ci=email).count():
-            errors.add_error(localizer.translate(_(
-                "We already have a user with this email.")),
-                ErrorTypes.EXISTING_EMAIL,
-                HTTPConflict.code)
+            if not discussion.preferences['generic_auth_errors']:
+                errors.add_error(localizer.translate(_(
+                    "We already have a user with this email.")),
+                    ErrorTypes.EXISTING_EMAIL,
+                    HTTPConflict.code)
+            else:
+                errors.add_error(localizer.translate(_(
+                    generic_error_message)),
+                    ErrorTypes.EXISTING_EMAIL,
```

Comment:
This, in fact, reveals a lot to the front-end. Must add a new enumType to `ErrorTypes`, perhaps `GENERIC`. And the front-end will have to deal with that case. I can do the front-end @mohamedzayed92 .